Change WiX convert error messages to information messages where appropriate

MarkStega opened the pull request.

robmen commented on src/wix/wix/Properties/launchSettings.json:
```diff
@@ -0,0 +1,9 @@
+{
```
I don't think we should add launchSettings.json generally. :)

MarkStega replied:
It's a temporary addition and will be removed before the final addition.

robmen commented on README.md:
```diff
@@ -7,3 +7,40 @@
 # WiX Toolset v4
 
 This repository contains the WiX Toolset v4 codebase.
+
```
These changes don't go with the title of the PR. Is this about doc or were other files supposed to go in?

MarkStega replied:
Lots of other files are going in, this was just a convenient addition (and I was hoping with draft status you wouldn't review, but that's OK)

robmen replied:
Ahh, my bad. I didn't notice it was still in Draft. I'll watch for that now since it doesn't look like there is anything in the email from GitHub to denote the PR is still in draft.

robmen commented on README.md:
```diff
@@ -7,3 +7,40 @@
+| MSVC v141 - VS 2017 C++ ARM64 build tools (v14.16) | |
+| MSVC v142 - VS 2017 C++ x64/x86 build tools (v14.16) | |
+| MSVC v142 - VS 2019 C++ ARM64 build tools (v14.29-16.11) | Required for WixE2E.WixE2EFixture.CanBuildPackageWithNativeWindowsApp |
+| MSVC v142 - VS 2019 C++ x64/x86 build tools (v14.29-16.11) | Required for WixE2E.WixE2EFixture.CanBuildPackageWithNativeWindowsApp |
```
These three lines should no longer be necessary with #98

MarkStega replied:
Correct, and they are already fixed locally.

MarkStega commented:
1) For some reason, after getting an update to VS2022 17.1 preview there was an issue finding the .Net 6.0.101 SDK; I added a global.json file top the root that excludes pre-release sdk's

2) The readme,md is updated to have the start of 'Developing WiX' infomation

3) WixConverter.cs now has a method 'OnInformation'. It displays an infomation level message prefixed with [Converted] to indicate that the construct in question was updated. I changed the 'Errors' member to 'Messages' but did not go through the test fixtures to change the name of the locals. I will if desired. The methodology of deciding whether to replace OnError with OnInformation was that if a transform was performed then OnInformation was appropriate.

rseanhall commented:
1. We generate a global.json in `src` which already excludes prerelease versions: https://github.com/wixtoolset/wix4/blob/212a7179408758b27fea5a12274f739a5205b2d7/src/internal/SetBuildNumber/SetBuildNumber.proj#L41
We haven't typically been interested in spending the time to support preview versions of VS or .NET Core, but there have been breaking changes in the past that we would have noticed earlier if someone had been brave enough to use the preview version. We would be interested in the details of this problem before solving it this way, and would prefer to do that in a separate PR.

2. We would prefer to take this in a separate PR so that the doc changes and conversion changes can be reviewed and merged independent of each other.

MarkStega commented:
Sean,

Not a problem -- I droped the global.json from this PR and will do a build demonstrating what I saw that lead me to the addition of same. I am building using the release version of VS2022.

robmen commented:
Sorry, I missed (or didn't get) the notice that this PR was out of draft. I've rebased it to current and did a bit of clean-up in PR #135.

Thanks for doing the work to bring this about.

rseanhall commented:
I don't think I've ever gotten a notification that a PR was marked ready for review. I would have expected that review request to have generated a notification, though.
